scanner/src/test/kotlin/scanners/scancode/ScanCodeResultParserTest.kt

```diff
@@ -654,12 +654,13 @@ class ScanCodeResultParserTest : FreeSpec({
         "fixup remaining exceptions" {
             associateLicensesWithExceptions(
                 listOf(
-                    LicenseFinding("Apache-2.0", TextLocation("file", 1)),
-                    LicenseFinding("LLVM-exception", TextLocation("file", 100))
+                    LicenseFinding("GPL-2.0-only", TextLocation("file", 1)),
+                    LicenseFinding("389-exception", TextLocation("file", 100))
                 )
             ) should containExactlyInAnyOrder(
-                LicenseFinding("Apache-2.0", TextLocation("file", 1)),
-                LicenseFinding("Apache-2.0 WITH LLVM-exception", TextLocation("file", 100))
+                LicenseFinding("GPL-2.0-only", TextLocation("file", 1)),
+                LicenseFinding("GPL-2.0-only WITH 389-exception", TextLocation("file", 100)),
+                LicenseFinding("GPL-2.0-or-later WITH 389-exception", TextLocation("file", 100))
```

Comment:
This makes me wonder if it would be better to associate any matching license finding regardless of the line to the exception. Otherwise this mechanism introduces licenses which were not detected anywhere, e.g. GPL-2.0-or-later in this case, or when I look at the mapping in the previous commit it could also introduce different versions of GPL licenses. And if no matching license is detected in the whole file it might be better to create an issue than to make up an association, at least if it is ambiguous, because it needs a manual review anyway.

Reply:
To simplify things, how about associating remaining exceptions with just `NOASSERTION` in general, instead of all applicable licenses?

Reply:
Was first thinking associating "orphan" exceptions to the root/main license but then I realized that finding them in scans would be difficult. 👍 from me to associate "orphan" exceptions with `NOASSERTION` - this should make it easy for ORT users to write a Evaluator rule requiring all exceptions in-combination with a `NOASSERTION` need to be investigated and resolved.